c2rust-analyze/src/pointer_id.rs

```diff
@@ -302,18 +401,29 @@ impl<T> GlobalPointerTable<T> {
 impl<T> Index<PointerId> for GlobalPointerTable<T> {
     type Output = T;
     fn index(&self, id: PointerId) -> &T {
-        assert!(id.is_global());
         &self.0[id.index()]
     }
 }
 
 impl<T> IndexMut<PointerId> for GlobalPointerTable<T> {
     fn index_mut(&mut self, id: PointerId) -> &mut T {
-        assert!(id.is_global());
         &mut self.0[id.index()]
     }
 }
 
+pub struct PointerRange {
+    base: u32,
+    len: u32,
+}
+
+impl PointerRange {
+    pub fn contains(&self, ptr: PointerId) -> bool {
+        // If `ptr.index() < self.base`, the subtraction will wrap to a large number in excess of
```

Comment:
Does this require that `base + len` not overflow?

Reply:
Update: I see above that you check for this. Maybe add a comment here?

Reply:
Added a comment in b68e56876

I think this `contains` method actually works when `base + len` overflows, but some of the pointers near the top of the range would be unrepresentable as a `PointerId`.